jans-cedarling/cedarling/src/jwt/decoding_strategy/key_service.rs

```diff
@@ -9,32 +9,101 @@ mod error;
 mod openid_config;
 
 pub use error::KeyServiceError;
-use jsonwebtoken::jwk::JwkSet;
+use jsonwebtoken::jwk::Jwk;
 use jsonwebtoken::DecodingKey;
 pub(crate) use openid_config::*;
 use reqwest::blocking::Client;
+use serde::Deserialize;
 use std::collections::HashMap;
 use std::sync::Arc;
+use std::thread::sleep;
+use std::time::Duration;
 
+/// A wrapper around `reqwest::blocking::Client` providing HTTP request functionality with retry logic.
+///
+/// The `HttpClient` struct allows for sending GET requests with a retry mechanism that attempts to
+/// fetch the requested resource up to a maximum number of times if an error occurs.
+pub struct HttpClient(reqwest::blocking::Client);
+
+impl HttpClient {
+    pub fn new() -> Self {
+        Self(reqwest::blocking::Client::new())
+    }
+
+    /// Sends a GET request to the specified URI with retry logic.
+    ///
+    /// This method will attempt to fetch the resource up to `MAX_RETRIES` times, with an increasing delay
+    /// between each attempt. The delay duration is adjusted for testing and non-testing environments.
+    fn get(&self, uri: &str) -> Result<reqwest::blocking::Response, KeyServiceError> {
+        const MAX_RETRIES: u32 = 3;
```

Comment:
Instead of constants like these, it's usually better to have them as struct members.

That would also provide a path to making `#cfg[(test])` unnecessary here.